modules/puppetca/puppetca_puppet_cert.rb

```diff
@@ -19,6 +19,14 @@ def clean certname
 
     # list of all certificates and their state/fingerprint
     def list
+      if to_bool(::Proxy::PuppetCa::Plugin.settings.puppet_6, false)
```

Comment:
I think it's better to inject an instance of a puppet 6 specific provider class here: 
https://github.com/theforeman/smart-proxy/blob/9d4605334ad7fec1b2e344cba15f15567db9c00d/modules/puppetca/plugin_configuration.rb#L10